cpp/tests/column/column_view_shallow_test.cpp

```diff
@@ -0,0 +1,219 @@
+#include <cudf/types.hpp>
+#include <cudf/utilities/traits.hpp>
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_utilities.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/cudf_gtest.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <thrust/iterator/counting_iterator.h>
+
+#include <memory>
+#include <type_traits>
+
+template <typename T>
+std::unique_ptr<cudf::column> example_column()
+{
+  // fixed_width, dict, string, list, struct
+  if constexpr (cudf::is_fixed_width<T>()) {
+    auto begin = thrust::make_counting_iterator(1);
+    auto end   = thrust::make_counting_iterator(16);
+    return cudf::test::fixed_width_column_wrapper<T>(begin, end).release();
+  } else if constexpr (cudf::is_dictionary<T>()) {
+    return cudf::test::dictionary_column_wrapper<std::string>(
+             {"fff", "aaa", "ddd", "bbb", "ccc", "ccc", "ccc", "", ""}, {1, 1, 1, 1, 1, 1, 1, 1, 0})
+      .release();
+  } else if constexpr (std::is_same_v<T, std::string> or std::is_same_v<T, cudf::string_view>) {
+    return cudf::test::strings_column_wrapper(
+             {"fff", "aaa", "ddd", "bbb", "ccc", "ccc", "ccc", "", ""})
+      .release();
+  } else if constexpr (std::is_same_v<T, cudf::list_view>) {
+    return cudf::test::lists_column_wrapper<int>({{1, 2, 3}, {4, 5}, {}, {6, 7, 8}}).release();
+  } else if constexpr (std::is_same_v<T, cudf::struct_view>) {
+    auto begin    = thrust::make_counting_iterator(1);
+    auto end      = thrust::make_counting_iterator(16);
+    auto member_0 = cudf::test::fixed_width_column_wrapper<int32_t>(begin, end);
+    auto member_1 = cudf::test::fixed_width_column_wrapper<int32_t>(begin + 10, end + 10);
+    return cudf::test::structs_column_wrapper({member_0, member_1}).release();
+  }
+  return {};
+}
+
+template <typename T>
+struct ColumnViewShallowTests : public cudf::test::BaseFixture {
+};
+
+using AllTypes = cudf::test::Concat<cudf::test::AllTypes, cudf::test::CompoundTypes>;
+TYPED_TEST_CASE(ColumnViewShallowTests, AllTypes);
+
+// Test for fixed_width, dict, string, list, struct
+// column_view, column_view = same hash.
+// column_view, make a copy = same hash.
+// column_view, copy column = diff hash
+// column_view old, update data + new column_view     = same hash.
+// column_view old, add null_mask + new column_view   = diff hash.
+// column_view old, update nulls + new column_view    = same hash.
+// column_view old, set_null_count + new column_view  = same hash.
+//
+// column_view, diff column     = diff hash.
+// column_view, sliced[0, size) = same hash (for split too)
+// column_view, sliced[n:)      = diff hash (for split too)
+// column_view, bit_cast        = diff hash
+//
+// mutable_column_view, column_view = same hash
+// mutable_column_view, modified mutable_column_view = same hash
+//
+// update the children column data  = same hash
+// update the children column_views = diff hash
+
+TYPED_TEST(ColumnViewShallowTests, shallow_hash)
```

Comment:
Rather than a single large test that needs a comment to describe the different cases, could you split this into multiple small tests? It shouldn't be much slower since these are all small data, and it would make it easier to read/debug in the future if a specific test failed.

Reply:
since this test directly calls EXPECT_*, test will point out the location of error. So, it will easier to debug.

For easier reading, it is split into multiple tests.